pkg/kubeclient/kubelet_client.go

```diff
@@ -258,6 +266,29 @@ func (client *KubeletClient) GetKubeletThresholds(ip, nodeName string) ([]evicti
 	return thresholds, nil
 }
 
+func (client *KubeletClient) GetCPUThrottlingMetrics(ip, nodeName string) (map[string]*dto.MetricFamily, error) {
+	data, err := client.ExecuteRequest(ip, nodeName, cadvisorPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var parser expfmt.TextParser
+	parsed, err := parser.TextToMetricFamilies(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(parsed) < 1 {
+		return nil, err
```

Comment:
This is no data and no error situation. `err` would always be nil here, because function returns on `err != nil` condition above. Updated to nil here, for better readability.